### ISSUE TITLE
Question+Suggestion: There's a lot of bloat that's still enabled, could you implementing adding brave browser launch flags?

### ISSUE THREAD

**Issue by BubbleSquish** [MulesGaming/brave-debloatinator#3]
Close

**Comment by Twig6943** [comment]
@BubbleSquish could you post an all-in-one flag for debullshitiniation?

**Comment by Twig6943** [comment]
FYI; Chromium supports the ability to save launch flags to a config file please see

https://wiki.archlinux.org/title/Chromium#Configuration
https://github.com/Twig6943/dotfiles/blob/main/chromium/cromite/cromite-flags.conf

**Comment by Twig6943** [comment]
> My main concern here is getting brave to launch with the flags applied. On linux I should just be able to modify the .desktop file to add launch flags, but for windows I can't really do that. I could edit the desktop shortcut but that wouldn't apply the flags when launching Brave any other way.
> 
> > FYI; Chromium supports the ability to save launch flags to a config file please see
> > https://wiki.archlinux.org/title/Chromium#Configuration
> > https://github.com/Twig6943/dotfiles/blob/main/chromium/cromite/cromite-flags.conf
> 
> As far as I can tell this is exclusive to Arch Linux and doesn't exist on any other platforms so I'm a bit hesitant to use the config file.

Nope its not exclusive to arch linux. I'm pretty sure this works with all linux distros. idk about mac and windows tho (somebody pls test)

**Comment by BubbleSquish** [comment]
> My main concern here is getting brave to launch with the flags applied. On linux I should just be able to modify the .desktop file to add launch flags, but for windows I can't really do that. I could edit the desktop shortcut but that wouldn't apply the flags when launching Brave any other way.
> 
> > FYI; Chromium supports the ability to save launch flags to a config file please see
> > https://wiki.archlinux.org/title/Chromium#Configuration
> > https://github.com/Twig6943/dotfiles/blob/main/chromium/cromite/cromite-flags.conf
> 
> As far as I can tell this is exclusive to Arch Linux and doesn't exist on any other platforms so I'm a bit hesitant to use the config file.

You're able to do this with most Linux distros, but the method for doing it with Windows is a little more complicated, but still pretty much basically the same. Just remember... Flags that apply to Linux don't always apply to Windows, and vise versa - simply because the feature isn't there yet for whichever operating system.

Simply put, for Windows, you can implement launch flags via registry. So basically, you can add the launch flags to your .reg file in your code. Which is mildly convenient since you're already doing something like that but for policies.

For almost all Linux distros. You can do the `Local State` method... Which is a little risky because you're modifying the users personal chromium profile, but it's literally just a text file with some gibberish related to their profile and the features enabled and then the flags they have enabled. So, you could just `echo` the debullshitter stuff in there. Or do the .desktop file, which I hate, don't do it. Or you can literally do `$which brave` (Consider nightly and beta as well) and then `echo` launch flags in their brave browser launcher, which is always a script and not an executable.

For nightly, I believe there's two fucking launchers called brave-nightly and brave-nightly-launcher... Which I haven't bothered to understand.

---

NOTE: There's policies that seem to be unable to be toggled off in:
brave://policy/logs

Notice `Active Variations`

Good Lord.... Anyway, I haven't figured out a way to toggle that shit off like the Ads related stuff and the other experimental surveillance features. This makes me believe that no matter what you do with your policies, they are still executing and communicating with their Rewards/VPN/AI/Ads bullshit as these `Active Variations` are only experimental and yet to be implemented. I haven't figured out how to stop them.

**Comment by BubbleSquish** [comment]
> @BubbleSquish could you post an all-in-one flag combination (flags to disable max amount of shit as possible) for debullshitiniation?

Yeah sure... Just give me a little because I have flags I am just fucking around with and I have to compile all the ones that relate to all debullshitting into a list.

**Comment by Twig6943** [comment]
> > @BubbleSquish could you post an all-in-one flag combination (flags to disable max amount of shit as possible) for debullshitiniation?
> 
> Yeah sure... Just give me a little because I have flags I am just fucking around with and I have to compile all the ones that relate to all debullshitting into a list.

alright alright bet

**Comment by Twig6943** [comment]
@BubbleSquish soo its been a while

**Comment by BubbleSquish** [comment]
> [@BubbleSquish](https://github.com/BubbleSquish) soo its been a while

My bad, I have been working on insurance and doctor appointments (I am fine I am just switching up things because of outdated info conflicting with my current)

What OS are you running? Do you want an automated script example or...? I'll just give you the breakdown until you specify what exactly you are wanting.

First, undo the policies set by this script by backing up and removing your current .json or reg entries for .

Reason: Due to policies, it could prevent your ability to actually have your changes take effect in between your customizations... So that's why you're simply backing it up, you'll re-apply it later.

So... Depending on OS, simply undo:

https://github.com/MulesGaming/brave-debullshitinator/blob/main/policies.json

or:

https://github.com/MulesGaming/brave-debullshitinator/blob/main/brave_debullshitinator-policies.reg

go to [brave://flags](url)

search in the search bar:

`brave-ads`
They trick you with this one, there's brave-**adblock** and brave-**ads**... You want to target brave-**ads**
`brave-rewards`
`brave-ai`
`brave-wallet`
(don't forget `native-brave-wallet`)
`brave-news`
`brave-ntp`

Simply set them all to disabled. (I'm not playing, it's literally just setting it to disabled... I've read their descriptions and yes, disabled is the correct configuration)

I am playing around with:
[EnableExperimentalPolicies](https://support.brave.com/hc/en-us/articles/360039248271-Group-Policy?policy=EnableExperimentalPolicies)
Wondering what the fuck it does, and I am trying to see what else there is.

Also. @MulesGaming ... It looks like they removed `NewTabPageLocation` from literally doing anything. your configuration is now known as an invalid setting. However, it's still a policy... Just, invalid. The hard-locked ntp location isn't a thing anymore. You have to manually configure it in settings.

WHICH MEANS THE BITCH IS STILL RUNNING IN THE BACKGROUND AAAAAAAAAAAAAAAAAAAAAAHHHHHHH!!!!!!!!!!!!!

HELP!

Oh, and all this stuff was still running in the background regardless of what you can currently do on the frontend... We're going to have to compile our own brave. I can't use this shit anymore. They change it up and screw up literally everything constantly.

We should call our brave...

Scare.

Or...

Archangel Michael

Fuck yeah.

**Comment by BubbleSquish** [comment]
Just remember, I am kind of knew to github and coding in general, so don't take everything I say as factual.

😅

**Comment by MulesGaming** [comment]
I've disabled all the flags you've mentioned but nothing except `brave-ntp-search-widget` and the ai related ones seems to have actually disabled anything. Rewards, wallet, and new tab page ads are still active.

**Comment by BubbleSquish** [comment]
> I've disabled all the flags you've mentioned but nothing except `brave-ntp-search-widget` and the ai related ones seems to have actually disabled anything. Rewards, wallet, and new tab page ads are still active.

....There should be some processes or subprocess that's active that's running the crypto and lovely ads.
I'm not sure though, because whoever created javascript really wants web3. It's not likely able to be changed without making our changes and recompiling and hopefully without having to do much.

But there's the issue with giving what we personally believe is a common annoyance, that of which is bullshit. Removing it, obviously. Do not wastefully discourage yourself. I'm full of debullshit and will look into it.

**Comment by BubbleSquish** [comment]
> My main concern here is getting brave to launch with the flags applied. On linux I should just be able to modify the .desktop file to add launch flags, but for windows I can't really do that. I could edit the desktop shortcut but that wouldn't apply the flags when launching Brave any other way.
> 
> > FYI; Chromium supports the ability to save launch flags to a config file please see
> > https://wiki.archlinux.org/title/Chromium#Configuration
> > https://github.com/Twig6943/dotfiles/blob/main/chromium/cromite/cromite-flags.conf
> 
> As far as I can tell this is exclusive to Arch Linux and doesn't exist on any other platforms so I'm a bit hesitant to use the config file.

Previously mentioned "local state method"
https://github.com/lepras/BraveBrowserDebloater/blob/main/BraveBrowserDebloater.sh
Edit the `Local State` file found in the users Brave folder and just put in your env var
BRAVE_DBLSHITR_FLGS='cat /path/to/wherever/that/will/be/DBLSHITR_FLAGS.conf'
call upon your wanted flags configuration
use /etc/environment to lock it in place globally using colorful mystery code
``

Random thing I found:
https://github.com/foxium-browser/foxium

**Comment by MulesGaming** [comment]
Modifying the local state & preferences files seems like it should work, and would allow more de-bloating options as well because were able to modify settings directly, but even if we do apply flags this way though there is still the issue of some of the flags not actually doing anything. We could just modify the settings however that still leaves remnants of the "features" that we're trying to disable. The main reason I went with group policies when first designing this is that is nearly whipped also visible parts of the bloat out of the browser, which just changing settings doesn't do. I might look at changing the flags that do work, and maybe some settings in the future, but unless you can figure out how to make flags disable everything, I don't think this can fully replace group policies. 

_For future reference the paths to the Brave user folder is `C:\Users\{user}\AppData\Local\BraveSoftware\Brave-Browser` on windows and `/home/{user}/.config/BraveSoftware/Brave-Browser` on linux._
  
  
> Random thing I found:
> https://github.com/foxium-browser/foxium

This seems cool, but if we were to fork brave, we would also be left with the responsibility of maintaining it, which is a lot for a browser which revives minor updates multiple times a week

**Comment by BubbleSquish** [comment]
@MulesGaming 

I never wanted to replace the policy stuff, it's just...

...At which part now are we at where flags aren't doing much and neither is policies due to the "HELLO! WOULD YOU LIKE TO OPT INTO STATISTICS", "BRAVE CRASHED SEND US YOUR SHIT", and "oops I somehow accidently clicked "aight bet" because you're yelling at me under right under my bookmark bar through rude banner notifications". Then, there's also this whole NTP issue that's refusing to comply with our demands to completely flush.

**Comment by BubbleSquish** [comment]
> This seems cool, but if we were to fork brave, we would also be left with the responsibility of maintaining it, which is a lot for a browser which revives minor updates multiple times a week

Well... Heh! Let's do it.

We do not like bullshit, imagine... Brave but without crypto and ads.... It's almost like the devs knew this challenge would come, will we be sued by johovascript? I would love a web browser that does exactly what brave does.... Without all the things that the forum people are claiming can be easily be disabled despite what we seem to be confusingly running into on github of all places.

I mean... you can Google or "Bravely search it" and everyone who's complaining about it is being ass blasted by keyboard warriors for claiming the issue doesn't exist because you can pressy the happy-toggle-off button.

**Comment by MulesGaming** [comment]
> ...At which part now are we at where flags aren't doing much and neither is policies due to the "HELLO! WOULD YOU LIKE TO OPT INTO STATISTICS", "BRAVE CRASHED SEND US YOUR SHIT", and "oops I somehow accidently clicked "aight bet" because you're yelling at me under right under my bookmark bar through rude banner notifications". Then, there's also this whole NTP issue that's refusing to comply with our demands to completely flush.

I agree. If there is a proper way to modify Brave to fully disable this I'll happily add it to the repo. If we can genuinely get flags to work they seem like they could fix stuff like this.

> Well... Heh! Let's do it.
> 
> We do not like bullshit, imagine... Brave but without crypto and ads.... It's almost like the devs knew this challenge would come, will we be sued by johovascript? I would love a web browser that does exactly what brave does.... Without all the things that the forum people are claiming can be easily be disabled despite what we seem to be confusingly running into on github of all places.
> 
> I mean... you can Google or "Bravely search it" and everyone who's complaining about it is being ass blasted by keyboard warriors for claiming the issue doesn't exist because you can press the happy-toggle-off button and issue gone. (it's not, are these people `for` real?)

I'd support a de-bloated Brave fork but I don't really want the responsibility or time commitment of maintaining a fork of a browser. Even if you just look at [Cromite](github.com/uazo/cromite) there are multiple commits a day.

**Comment by Twig6943** [comment]
Meeh tbh people should just use cromite instead [Cromite](github.com/uazo/cromite) its the only foss chromium non-bloated browser that still has a built-in adblocker

**Comment by BubbleSquish** [comment]
> Meeh tbh people should just use cromite instead [Cromite](github.com/uazo/cromite) its the only foss chromium non-bloated browser that still has a built-in adblocker

Do it even matter though?

Right now I am just using brave because of its superior force-dark implementations (which is mostly just up-to-date chrome-dev features with its own twist)

**Comment by MulesGaming** [comment]
> Meeh tbh people should just use cromite instead [Cromite](github.com/uazo/cromite) its the only foss chromium non-bloated browser that still has a built-in adblocker

Yeah, I'm using Cromite as my browser on my phone and it's good. I haven't really used it on desktop though. It's built in adblocker isn't as good as Brave's (but Brave's is the only built-in adblocker I've seen that gets anywhere close to uBlock origin), but that's not really a big deal because uBlock Origin exists.

**Comment by BubbleSquish** [comment]
Closed.
